src/material-experimental/mdc-slider/slider.ts

```diff
@@ -22,13 +75,229 @@ import {SliderAdapter} from './slider-adapter';
+
+  /** Event emitted when a slider thumb moves. */
+  @Output() readonly input: EventEmitter<MatSliderChange> = new EventEmitter<MatSliderChange>();
+
+  /** Event emitted when the slider thumb starts being dragged. */
+  @Output() readonly dragStart: EventEmitter<MatSliderInteraction>
+    = new EventEmitter<MatSliderInteraction>();
+
+  /** Event emitted when the slider thumb stops being dragged. */
+  @Output() readonly dragEnd: EventEmitter<MatSliderInteraction>
+    = new EventEmitter<MatSliderInteraction>();
+
+  /** Whether this is a ranged slider. */
+  get isRange(): boolean { return this._inputs.length === 2; }
 
-  // tslint:disable-next-line:no-unused-variable
+  /** Instance of the MDC slider foundation for this slider. */
   private _foundation = new MDCSliderFoundation(new SliderAdapter());
+
+  /** Whether the foundation has been initialized. */
+  _initialized: boolean = false;
+
+  /** The string representation of the start thumbs value. */
+  _startValueIndicatorText: string;
+
+  /** The string representation of the end thumbs value. */
+  _endValueIndicatorText: string;
+
+  /** Used to keep track of & render the active & inactive tick marks on the slider track. */
+  get tickMarks(): TickMark[] { return this._tickMarks; }
+  set tickMarks(v: TickMark[]) {
+    this._tickMarks = v;
+    this._cdr.markForCheck();
+  }
+  private _tickMarks: TickMark[];
+
+  constructor(
+    private _cdr: ChangeDetectorRef,
+    private readonly _elementRef: ElementRef<HTMLElement>,
+    private readonly _platform: Platform,
+    @Inject(DOCUMENT) private readonly _document: any) {}
+
+  ngAfterViewInit() {
+    this._foundation.init();
+    if (this._platform.isBrowser) {
+      this._foundation.layout();
+    }
+  }
+
+  ngOnDestroy() {
+    if (this._platform.isBrowser) {
+      this._foundation.destroy();
+    }
+  }
+
+  /** Gets the current value of given slider thumb. */
+  getValue(thumb: Thumb): number {
+    return thumb === Thumb.START
+      ? this._foundation.getValueStart()
+      : this._foundation.getValue();
+  }
+
+  /** Sets the value of a slider thumb. */
+  setValue(value: number, thumb: Thumb): void {
+    thumb === Thumb.START
+      ? this._foundation.setValueStart(value)
+      : this._foundation.setValue(value);
+  }
+
+  /** Returns the injected document if available or fallback to the global document reference. */
+  _getDocument(): Document {
+    return this._document || document;
+  }
+
+  /**
+   * Returns the defaultView of the injected document
+   * if available or fallback to global window reference.
+   */
+  _getWindow(): Window {
+    return this._document.defaultView || window;
+  }
+
+  /** Gets the hosts native HTML element. */
+  _getHostElement(): HTMLElement {
+    return this._elementRef.nativeElement;
+  }
+
+  /** Gets the slider thumb input of the given thumb. */
+  _getInput(thumb: Thumb): MatSliderThumb {
```

Comment:
What are all of these internal methods used for?

Reply:
These are just some getters for the `SliderAdapter`

Reply:
Ah gotcha, I guess the class that uses them is not in this PR

Reply:
I think maybe I'll have some organization comments when you send out the PR with the adapter implementation, but for now this is fine